packages/lib-utils/src/app/api-discovery/discovery-cache.ts

```diff
@@ -1,23 +1,51 @@
 import { consoleLogger } from '@monorepo/common';
+import * as _ from 'lodash-es';
+import { getReferenceForModel } from '../../k8s/k8s-utils';
 import type { DiscoveryResources } from '../../types/api-discovery';
+import type { K8sModelCommon } from '../../types/k8s';
 
 const SDK_API_DISCOVERY_RESOURCES_LOCAL_STORAGE_KEY = 'sdk/api-discovery-resources';
 
-export const cacheResources = (resources: DiscoveryResources) => {
+const mergeByKey = (prev: K8sModelCommon[], next: K8sModelCommon[]) =>
+  Object.values(_.merge(_.keyBy(prev, getReferenceForModel), _.keyBy(next, getReferenceForModel)));
+
+export const cacheResources = (resources: DiscoveryResources[]) => {
+  let allResources;
+  try {
+    allResources = [
+      ...[JSON.parse(localStorage.getItem(SDK_API_DISCOVERY_RESOURCES_LOCAL_STORAGE_KEY) || '{}')],
+      ...resources,
+    ].reduce((acc, curr) => {
+      const { models, configResources, clusterOperatorConfigResources, ...rest } = curr || {};
```

Comment:
For some reason lodash messes merge and if `models`, `configResources` or `clusterOperatorConfigResources` were in merged object they were rewritten and only last ones were used.

Reply:
@karelhala hm... I was hoping for a cleaner implementation of the cache merge... Perhaps this is due to the fact that lodash mutates the target object passed in. Have you tried cloning the objects before merging with spread?

Reply:
The problem is that `merge` from lodash is merging the data over keys and if we merge multiple arrays together they get overwritten by the second (because arrays have same keys). That's why I had to come up with the `mergeByKey` just for the arrays.

If I kept the `models`, `configResources` or `clusterOperatorConfigResources` in the current object the spread was ignored. That's because lodash is performing the merge in recursive manner, so when the reduce finishes it uses the later value (because these are nested and lodash is doing some perfomance goodies on top of this). We could change the `reduce` arguments to
```JS
.reduce(
  {
    models: accModels,
    configResources: accConfigResources,
    clusterOperatorConfigResources: accClusterOperatorConfigResources,
    ...acc
  } = {},
  { 
    models: currModels,
    configResources: currConfigResources,
    clusterOperatorConfigResources: currClusterOperatorConfigResources,
    ...curr
  } ={}
)
```
Not sure if this would make things easier to read. I can try to play around with `mergeWith` and write a bit more complex compare function.